src/exception/exception-manifest/types.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { DeepReadonly } from 'deep-freeze';
+import { FollowUp, Specimen, Treatment } from '../../clinical/clinical-entities';
+
+export const MissingEntityExceptionType = 'MissingEntity';
+
+export const ProgramExceptionType = 'ProgramProperty';
+
+export const EntityPropertyExceptionType = 'EntityProperty';
```

Comment:
Not requesting a change. Just adding 2cents.

If you structure these as an object like:
```ts
export const ExceptionType = {
	missingEntity: 'MissingEntity',
	entity: 'EntityProperty',
	program: 'ProgramProperty',
}
```
This means everywhere you manually put the string 'MissingEntity' you can do `ExceptionType.missingEntity`.
Then if you ever need to change it, you can change it in this object alone, and there will be no dangling references to a previous string.

Now we build some types to use throughout....

currently this is valid: `ExceptionType.missingEntity = 'the cake is a lie';`
now we've messed up our object.
this is also valid: `ExceptionType.fakeEntityForTesting = 123`
also messed up our object
How do we lock this down to just the values we declare originally? `as const`

```ts
export const ExceptionType = {
	missingEntity: 'MissingEntity',
	entity: 'EntityProperty',
	program: 'ProgramProperty',
} as const
```
by using `as const` we narrow the object.
`ExceptionType.missingEntity = 'the cake is a lie';` this doesnt work. the property can only be 'MissingEntity'.
we also can't add values to the object.

Reply:
Excellent thank you, I will carry this over to the next PR